lib/twitter_cldr/utils/script_detector.rb

```diff
@@ -0,0 +1,81 @@
+# encoding: UTF-8
+
+# Copyright 2012 Twitter, Inc
+# http://www.apache.org/licenses/LICENSE-2.0
+
+module TwitterCldr
+  module Utils
+    class ScriptDetectionResult
+
+      attr_reader :scores
+
+      def initialize(scores)
+        @scores = scores
+      end
+
+      def best_guess
+        max_score = 0
+        max_script_name = nil
+
+        scores.each_pair do |script_name, score|
```

Comment:
Yes except - and you're going to think I'm crazy - `#each_pair` is more efficient. If we use `#max_by`, a new array is created for each key/value yield. This probably falls directly into the category of premature optimization, since the `scores` hash is likely to be quite small.